backend/apps/kloudust/lib/cmd/scripts/powerOpVM.sh

```diff
@@ -3,17 +3,50 @@
 # Params
 # {1} - Domain / VM name
 # {2} - The operation to perform
+# {3} - Optional wait time before checking state (in seconds)
 
-NAME="{1}"
+VM_NAME="{1}"
 POWER_OP="{2}"
+WAIT_TIME="${3:-10}"
 
 function exitFailed() {
     echo Failed
     exit 1
 }
 
-printf "Power operating $NAME to $POWER_OP\n"
-if ! virsh $POWER_OP $NAME; then exitFailed; fi
+function backgroundCheckAndStartIfNeeded() {
+    local vm="$1"
+    local wait="$2"
 
-printf "\n\nPower operation $POWER_OP successful on $NAME\n"
-exit 0
+    (
+        sleep "$wait"
+        STATE=$(virsh domstate "$vm" 2>/dev/null)
+        if [[ "$STATE" == "shut off" ]]; then
+            echo "$vm is still shut off after $wait seconds. Starting it manually..."
+            if ! virsh start "$vm"; then
+                echo "Failed to start $vm after reboot"
+            else
+                echo "$vm started manually after reboot"
+            fi
+        else
+            echo "$vm is in state: $STATE after $wait seconds. No action needed."
+        fi
+    ) &
+}
+
+echo "Power operating $VM_NAME to $POWER_OP"
+
+if [[ "$POWER_OP" == "reboot" ]]; then
+    if ! virsh reboot "$VM_NAME"; then
```

Comment:
Why double condition like this? Whatever the power op is you have to do it always. You just needed to add a check in case it was reboot.